scenario_player/utils/reclaim.py

```diff
@@ -0,0 +1,520 @@
+        )
+        return
+
+    expiration_block = BlockExpiration(100000000000000)
+    total_withdraw = WithdrawAmount(details.our_details.withdrawn + new_withdraw)
+    packed_withdraw = pack_withdraw(
+        canonical_identifier=CanonicalIdentifier(
+            chain_identifier=token_network.chain_id(),
+            token_network_address=token_network.address,
+            channel_identifier=channel["channel_identifier"],
+        ),
+        participant=to_canonical_address(address),
+        total_withdraw=total_withdraw,
+        expiration_block=expiration_block,
+    )
+
+    try:
+        token_network.set_total_withdraw(
+            given_block_identifier=current_confirmed_head,
+            channel_identifier=channel["channel_identifier"],
+            total_withdraw=total_withdraw,
+            expiration_block=expiration_block,
+            participant=to_canonical_address(address),
+            partner=to_canonical_address(details.partner_details.address),
+            participant_signature=LocalSigner(privkey).sign(packed_withdraw),
+            partner_signature=LocalSigner(partner_candidate.privkey).sign(packed_withdraw),
+        )
+    except InsufficientEth:
+        log.warning("Not enough ETH to withdraw", channel=channel)
+    else:
+        log.info("Withdraw successful", channel=channel, amount=new_withdraw)
+
+
+def withdraw_all(
+    address_to_candidate: Dict[Address, ReclamationCandidate],
+    account: Account,
+    eth_rpc_endpoint: URI,
+    contract_manager: ContractManager,
+    token_address: TokenAddress,
+) -> None:
+    """ Withdraws all tokens from all channels
+
+    For this to work, both channel participants have to be in ``reclamation_candidates``.
+    All tokens will be withdrawn to participant1, ignoring all balance proofs.
+    """
+    web3 = Web3(HTTPProvider(eth_rpc_endpoint))
+    chain_id = ChainID(web3.eth.chainId)
+    deploy = get_contracts_deployment_info(chain_id, RAIDEN_CONTRACT_VERSION)
+    assert deploy
+    assert account.privkey
+    token_network_address = _get_token_network_address(
+        token_address=token_address, web3=web3, privkey=account.privkey, deploy=deploy
+    )
+    token_network_deployed_at = BlockNumber(
+        deploy["contracts"][CONTRACT_TOKEN_NETWORK_REGISTRY]["block_number"]
+    )
+
+    # This assumes that reclamation will be done after the scenarios have
+    # executed, so we don't have to synchronize with the whole blockchain, just
+    # up to the last block used by the scenario run. Since we don't know the
+    # exact block, use the current confirmed head.
+    #
+    # This also assumes no RPC calls to pruned data will be performed, so the
+    # target_block is never updated!
+    confirmation_blocks = DEFAULT_NUMBER_OF_BLOCK_CONFIRMATIONS
+    latest_block = web3.eth.getBlock(BLOCK_ID_LATEST)
+    latest_block_number = latest_block["number"]
+    current_confirmed_head = BlockNumber(latest_block_number - confirmation_blocks)
```

Comment:
Caching the confirmed block, this saves RPC calls and streamlines the code (no need to do the same thing in multiple places).